### ISSUE TITLE
Is smbios-battery-ctl supported on the Dell Latitude 5420?

### ISSUE THREAD

**Issue by bmaupin** [dell/libsmbios#101]
Hi,

I have a Dell Latitude 5420. I set a custom charging mode using `smbios-battery-ctl`:

```
$ sudo smbios-battery-ctl --set-custom-charge-interval=50 55
Custom charge interval has been set to (50, 55)
$ sudo smbios-battery-ctl --set-charging-mode=custom
Charging mode has been set to: custom
$ sudo smbios-battery-ctl --get-charging-cfg
Charging mode: custom
Charging interval: (50, 55)
```

But then the battery proceeds to charge to 100% every time.

Things I've tried:

- Booting to BIOS to confirm the new charging mode; the settings were indeed reflected in the BIOS
- Discharging to below 50% and then trying again
- Rebooting

I'm using stock Ubuntu 20.04, and I installed the tool via the `smbios-utils` package in the normal Ubuntu apt repository.

1. Is `smbios-battery-ctl` supported on the Latitude 5420?
1. Are there any additional requirements to get `smbios-battery-ctl` to work on this machine?

In case it matters, I seem to be affected by #71 as per https://github.com/dell/libsmbios/issues/71#issuecomment-894463953

Thanks!

**Comment by bmaupin** [comment]
I just updated the BIOS to the latest version (1.10.0), hoping it would fix this issue, but at least while Ubuntu is running I'm still seeing the same behaviour.

I also noticed that the battery seems to charge to 100% even when the laptop is fully powered off, which leads me to believe this issue may be a hardware/BIOS issue. I'll keep testing and if need be I'll file a support ticket with Dell.

**Comment by bmaupin** [comment]
I went ahead and booted into BIOS and played with the settings directly there, because the BIOS screen also shows the charge percentage as well as the charging state (idle, charging, etc.)

I discovered that the problem seemed to be when I set the custom charging upper bound to 55, it doesn't seem to be respected. As soon as I set it to something higher than 55 it seems to work as expected. So I think it must be a BIOS bug.

Thanks!
